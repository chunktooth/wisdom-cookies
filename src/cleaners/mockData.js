/* exported variableName */
export const mockQuotes = [
  {
    'message': 'Mankind is unkind, man',
    'id': '5280'
  },
  {
    'message': 'Godluck and Goodspeed!',
    'id': '1234'
  }
];