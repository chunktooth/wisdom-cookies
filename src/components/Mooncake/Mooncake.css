.Mooncake {
  width: 50%;
}

.moon {
  width: 180px;
}

.newMoon {
  width: 100px;
  margin: auto;
}

.firstQuarter {
  width: 100px;
}

.fullMoon {
  width: 100px;
}

.lastQuarter {
  width: 100px;
}

.phase-name,
.phase-date {
  color: #FFFFFF;
}

.phase-date {
  font-size: .7em;
}

@media screen and (max-width: 320px) {
  .Mooncake {
    width: 100%;
  }
}