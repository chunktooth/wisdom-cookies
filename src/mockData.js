/* exported variableName */
export const mockWisdoms = [
  {
    'message': 'Mankind is unkind, man..',
    'id': '5280'
  },
  {
    'message': 'Godluck and Goodspeeeed!',
    'id': '1234'
  }
];

export const mockJar = [
  {
    'message': 'Mankind is unkind, man',
    'id': '5280'
  },
  {
    'message': 'Godluck and Goodspeed!',
    'id': '1234'
  }
];

export const mockMoonphases = { response: [
  { 
    timestamp: 1524433610, 
    dateTimeISO: "2018-04-22T21:46:50+00:00", 
    code: 1, 
    name: "first quarter"
  },
  {
    timestamp: 1525049954, 
    dateTimeISO: "2018-04-30T00:59:14+00:00", 
    code: 2, 
    name: "full moon"
  }]
};

export const mockMoonphasesArray = [{ response: [
  { 
    timestamp: 1524433610, 
    dateTimeISO: "2018-04-22T21:46:50+00:00", 
    code: 1, 
    name: "first quarter"
  },
  {
    timestamp: 1525049954, 
    dateTimeISO: "2018-04-30T00:59:14+00:00", 
    code: 2, 
    name: "full moon"
  }]
}]; 

export const mockMoonphase =  {
  date: "2018-04-22", 
  name: "first quarter"
};

export const mockWisdom = {
  'message': 'In Crust We Trust',
  'id': '1234'
};

export const mockCleanedMoons = [
  {
    date: "2018-04-22", 
    name: "first quarter"
  },
  { 
    date: "2018-04-30", 
    name: "full moon"
  }
];