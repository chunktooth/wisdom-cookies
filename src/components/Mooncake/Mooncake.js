import React from 'react';
import PropTypes from 'prop-types';
import newMoon from '../../images/cake-crumbles.png';
import firstQuarter from '../../images/first-quarter.png';
import fullMoon from '../../images/full-cake.png';
import lastQuarter from '../../images/last-quarter.png';
import './Mooncake.css';

export const Mooncake = ({ phase }) => {    
  return (
    <div className='Mooncake'>
      {
        phase.code === 0 &&
        <img src={newMoon} 
          className='new-moon' 
          alt="An empty plate of cake crumbles" />
      }
      {
        phase.code === 1 &&
        <img src={firstQuarter} 
          className='first-quarter' 
          alt="The left half of mooncake is eaten" />
      }
      {
        phase.code === 2 &&
        <img src={fullMoon} 
          className='full-moon' 
          alt="A full mooncake" />
      }
      {
        phase.code === 3 &&
        <img src={lastQuarter} 
          className='last-quarter' 
          alt="The right half of mooncake is eaten" />
      }
      <h3 className='phase-name'>{phase.name}</h3>
      <p className='phase-date'>on {phase.date}</p>
    </div>
  );
};

Mooncake.propTypes = {
  phase: PropTypes.object
};