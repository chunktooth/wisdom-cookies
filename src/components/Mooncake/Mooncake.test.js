import React from 'react';
import { Mooncake } from './Mooncake';
import { shallow } from 'enzyme';
import { mockMoonphase } from '../../mockData';

describe('Moons', () => {

  it('should match the snapshot', () => {
    let wrapper = shallow(<Mooncake 
      phase={mockMoonphase} />);
    expect(wrapper).toMatchSnapshot();
  });

});