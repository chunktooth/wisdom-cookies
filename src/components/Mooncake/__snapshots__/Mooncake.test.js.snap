// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Moons should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    phase={
        Object {
            "date": "2018-04-22",
            "name": "first quarter",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        false,
        false,
        false,
        <h3
          className="phase-name"
>
          first quarter
</h3>,
        <p
          className="phase-date"
>
          on 
          2018-04-22
</p>,
      ],
      "className": "Mooncake",
    },
    "ref": null,
    "rendered": Array [
      false,
      false,
      false,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "first quarter",
          "className": "phase-name",
        },
        "ref": null,
        "rendered": "first quarter",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "on ",
            "2018-04-22",
          ],
          "className": "phase-date",
        },
        "ref": null,
        "rendered": Array [
          "on ",
          "2018-04-22",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          false,
          false,
          false,
          <h3
            className="phase-name"
>
            first quarter
</h3>,
          <p
            className="phase-date"
>
            on 
            2018-04-22
</p>,
        ],
        "className": "Mooncake",
      },
      "ref": null,
      "rendered": Array [
        false,
        false,
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "first quarter",
            "className": "phase-name",
          },
          "ref": null,
          "rendered": "first quarter",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "on ",
              "2018-04-22",
            ],
            "className": "phase-date",
          },
          "ref": null,
          "rendered": Array [
            "on ",
            "2018-04-22",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
